### PR TITLE
Fixes #36319 - Add Promxox support

### PR DESCRIPTION
**Pull request by maximiliankolb**
I would like to add Proxmox to foreman-installer to simplify the installation process for users.

* PR in foreman-documentation: https://github.com/theforeman/foreman-documentation/pull/2142
* PR in puppet-foreman: https://github.com/theforeman/puppet-foreman/pull/1115

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
You need a migration for this. Something like this should work:
```sh
./bin/create-migration add-fog-proxmox config/foreman.migrations config/katello.migrations <<EOF
answers['foreman::plugin::proxmox'] ||= false
EOF
```

And then the test suite may need some modification to pass.

File: config/katello-answers.yaml [hunk 1]
```diff
@@ -61,6 +61,7 @@ foreman::plugin::leapp: false
 foreman::plugin::monitoring: false
```
Comment (sbernhard) [inline config/katello-answers.yaml:60]: Maybe add foreman::plugin::proxmox to foreman-answers.yaml, too? Its not releated to katello only.

Comment (maximiliankolb) [reply config/katello-answers.yaml:60]: Unless you mean something else, it already is: https://github.com/theforeman/foreman-installer/pull/845/files

**Comment by maximiliankolb** [comment]
Rebased to develop. I have no idea why the tests failed previously: https://ci.theforeman.org/blue/organizations/jenkins/foreman-installer-pr-test/detail/foreman-installer-pr-test/536/pipeline/70

I did not find any instructions such as "if you add a plug-in, you'll also have to extend the tests accordingly".

**Comment by maximiliankolb** [comment]
The pipeline failure looks like it's unrelated to my change. Unless told otherwise, I'd kindly ask someone to restart the pipeline some time _next week_.

**Comment by evgeni** [comment]
Yeah, we need a PostgreSQL module release with https://github.com/puppetlabs/puppetlabs-postgresql/commit/e6551ee4ed4796d1925c0828fc20a1dd9a9ef669 (or to pin stdlib/apt for now)

**Comment by sbernhard** [comment]
[test rpm]

**Comment by sbernhard** [comment]
[test]

**Comment by sbernhard** [comment]
[test foreman-installer]

**Review by ekohl — APPROVED** [review]
Code wise this is good and I'd like to get this in Foreman 3.7.

**Comment by ekohl** [comment]
[test foreman-installer]

**Comment by evgeni** [comment]
@ekohl This will fail until all modules support stdlib 9 or we pin it to < 9 and also all modules that try to pull 9 in (like puppetlabs/apt)

**Comment by ekohl** [comment]
Let's see if https://github.com/theforeman/foreman-installer/pull/857 helps.

**Comment by evgeni** [comment]
[test foreman-installer]

**Comment by evgeni** [comment]
@maximiliankolb this passes now, is there any other reason to keep this as Draft?

**Comment by maximiliankolb** [comment]
@evgeni Thanks for restarting the tests and merging!

**Comment by evgeni** [comment]
and here is the pick to 3.7 stable: https://github.com/theforeman/foreman-installer/pull/858

**Comment by maximiliankolb** [comment]
@evgeni Is there a cherry-pick necessary for https://github.com/theforeman/puppet-foreman/pull/1115 ?

**Comment by evgeni** [comment]
> @evgeni Is there a cherry-pick necessary for [theforeman/puppet-foreman#1115](https://github.com/theforeman/puppet-foreman/pull/1115) ?

nope, that is already in the release we ship in 3.7, the installer config was the only missing piece

**Pull request merged**